### DISCUSSION TITLE
Ajout des champs personnalisés

### DISCUSSION THREAD

**Discussion by dbarzin** [dbarzin/mercator#508] [Ideas]
La possibilité de définir des champs personnalisés va être ajouté dans Mercator !
Voici les principales fonctionnalités qui seront implémentées : 
- chaque table peut avoir des champs personnalisés
- cette personnalisation est accessible via le menu de configuration
- un champ personnalisé est composé : 
    - d'un nom, et d'une description,
    - d'un type (entier, date, chaîne de caractère, texte...)
    - d'une taille minimum et maximum
    - d'une expression régulière pour la saisie
    - d'un ordre d'apparition dans le formulaire (en cas de conflit l'ordre alphabétique sera appliqué)
- les champs personnalisés se trouvent toujours en dessous de la description de l’objet concerné
- certains champs actuels (à identifier) seront migrés en champs personnalisés.
- la recherche texte libre recherche aussi dans les champs personnalisés
- les champs personnalisés sont présents dans les rapports

Cette proposition sera développée courant 2025, elle est ouverte à commentaires.

**Comment by moiiom59** [comment]
Parfait, c'est une très bonne idée

**Comment by Neoliapaul** [comment]
Excellente nouvelle !
Je pense que migrer les champs qui ne sont pas "essentiels" en champ personnalisé est une bonne idée, après, à voir ce qu'il se passe si on renomme ou supprime un champ personnalisé, il faudrait qu'il y ait une alerte si des données sont présentes.
Est-ce que c'est prévu d'intégrer le type "liste à choix multiple" ?

**Reply by dbarzin** [reply]
Il sera possible de limiter les choix via une expression régulière

**Comment by moiiom59** [comment]
Ce que j'attends aussi c'est une plus fine gestion des droits.  Serait il possible de gerer le droits d'accès par tables ?
Par exemple le DPO aurait arccès au Volet RGPD, et la tables Informations. La secretaire SI aurait accès  à l'ecosysteme. Les tech SI aurraient accès pour certain a la partie Infrasstructure et d'autre Administrations etc...

**Reply by dbarzin** [reply]
Cette fonction est en place via la gestion des rôles, il est possible de définir de nouveaux rôles et de les assigner à des utilisateurs.
Ce qui n'est pas possible est d'assigner des droits par lignes d'une table (ex: une application, un fournisseur... en particulier).
